sql/plan/json_table.go

```diff
@@ -125,11 +163,13 @@ func (*JSONTable) CollationCoercibility(ctx *sql.Context) (collation sql.Collati
 
 // Expressions implements the sql.Expressioner interface
 func (t *JSONTable) Expressions() []sql.Expression {
+	// TODO: better to do nothing or return all exprs?
```

Comment:
Interesting... seems like you need to return all the expressions that are present in the data structure, otherwise they won't get processed correctly by the analyzer. Some test cases with having expressions that need resolution nested in the structure would be helpful, too. Seems like that might be a little tricky/messy to handle in `WithExpressions` to set the updated expressions.